### PR TITLE
[misc] Upgrade test and docker image to support python 3.10

### PR DESCRIPTION
**Pull request by qiao-bo**
Related issue = #3985 

This PR add test support for python 3.10 in Taichi. Note that torch-related executions are temporally disabled because it does not yet support 3.10.

CPU Linux 3.6 test is replaced with 3.10. For other OS we keep the current python version, but can be easily changed in the yml file.

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -483,6 +483,8 @@ def check_version():
         payload['python'] = 'cp38'
     elif python_version.startswith('3.9.'):
         payload['python'] = 'cp39'
+    elif python_version.startswith('3.10.'):
+        payload['python'] = 'cp310'
```
Comment (frostming) [inline python/taichi/lang/__init__.py:487]: Can be changed to a more concise form:
```python
python_version = sys.version_info
payload['python'] = f'cp{python_version[0]}{python_version[1]}'
```

**Comment by frostming** [comment]
Also need to update the `python_requires` range in `setup.py`

File: python/taichi/lang/__init__.py (hunk 1 above)
Comment (qiao-bo) [reply python/taichi/lang/__init__.py:487]: thanks, i believe you mean this?
```
python_version = platform.python_version().split('.')
payload['python'] = f'cp{python_version[0]}{python_version[1]}'
```

Comment (frostming) [reply python/taichi/lang/__init__.py:487]: `sys.version_info` is a tuple of version bits so we don't need to split it ourselves.

Comment (qiao-bo) [reply python/taichi/lang/__init__.py:487]: i see, thanks a lot, changed!

**Review by k-ye — APPROVED** [review]
Great job, LGTM!

File: .github/workflows/scripts/unix_test.sh [hunk 2]
```diff
@@ -20,7 +20,9 @@ fi
 python3 -m pip install dist/*.whl
 if [ -z "$GPU_TEST" ]; then
     python3 -m pip install -r requirements_test.txt
-    python3 -m pip install torch
+    if [[ $PY != *"3.10"* && $PY != *"py310"* ]]; then
```
Comment (frostming) [inline .github/workflows/scripts/unix_test.sh:23]: I suggest making a `requirements_torch.txt` which contains:
```
torch; python_version < '3.10'
```
Let `pip` decide whether to install based on the python version.

Comment (frostming) [reply .github/workflows/scripts/unix_test.sh:23]: Or more simple, via command line:

```
python3 -m pip install "torch; python_version < '3.10'"
```

**Comment by qiao-bo** [comment]
@frostming thanks a lot! changed made.

File: .github/workflows/scripts/win_test.ps1 [hunk 3]
```diff
@@ -7,10 +6,13 @@ ti diagnose
...
-    pip install torch==1.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
-} else {
-    pip install torch
+# TODO relax this when torch supports 3.10
+if (-not "$env:PYTHON".Contains("3.10")) {
+  if ("$env:TI_WANTED_ARCHS".Contains("cuda")) {
+      pip install torch==1.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
+  } else {
+      pip install torch
+  }
```
Comment (frostming) [inline .github/workflows/scripts/win_test.ps1:15]:
```suggestion
# TODO relax this when torch supports 3.10
if ("$env:TI_WANTED_ARCHS".Contains("cuda")) {
    pip install "torch==1.10.1+cu113; python_version < '3.10'" -f https://download.pytorch.org/whl/cu113/torch_stable.html
} else {
    pip install torch
}
```

**Pull request merged**